lightningd/plugin.c

```diff
@@ -90,8 +106,8 @@ static bool plugin_read_json_one(struct plugin *plugin)
 	toks = json_parse_input(plugin->buffer, plugin->used, &valid);
 	if (!toks) {
 		if (!valid) {
-			/* FIXME (cdecker) Print error and kill the plugin */
-			return io_close(plugin->stdout_conn);
+			plugin_kill(plugin, "Failed to parse JSON response");
```

Comment:
Log the actual output somewhere, for debugging?

Reply:
Should we roll that into `log_io`?